File: safari-600.7-branch/LayoutTests/http/tests/xmlhttprequest/resources/logout/resource.php
<?php
  header("Cache-Control: no-cache, no-store");

  if (!isset($_SERVER['PHP_AUTH_USER']) || !isset($_SERVER['PHP_AUTH_PW']) || $_SERVER['PHP_AUTH_USER'] != 'user' || $_SERVER['PHP_AUTH_PW'] != 'pass') {
   header('WWW-Authenticate: Basic realm="Please cancel this authentication dialog"');
   header('HTTP/1.0 401 Unauthorized');
   echo 'Authentication canceled';
   exit;
  } else {
   echo "User: {$_SERVER['PHP_AUTH_USER']}, password: {$_SERVER['PHP_AUTH_PW']}.";
  }
?>

File: LayoutTests/http/tests/xmlhttprequest/resources/logout/resource.py
#!/usr/bin/env python3
import base64
import os
import sys

sys.stdout.write(
    'Content-Type: text/html\r\n'
    'Cache-Control: no-cache, no-store\r\n'
)

credentials = base64.b64decode(os.environ.get('HTTP_AUTHORIZATION', ' Og==').split(' ')[1]).decode().split(':')
username = credentials[0]
password = ':'.join(credentials[1:])

if (username, password) == ('user', 'pass'):
    sys.stdout.write(
        '\r\n'
        'User: {}, password: {}.'.format(username, password)
    )

else:
    sys.stdout.write(
        'WWW-Authenticate: Basic realm="Please cancel this authentication dialog"\r\n'
        'status: 401\r\n'
        '\r\n'
        'Authentication canceled'
    )
